Watchman.Discord/Areas/Protection/Strategies/SpamPunishmentStrategy.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System;
+using Devscord.DiscordFramework.Framework.Commands.AntiSpam;
+using Devscord.DiscordFramework.Framework.Commands.AntiSpam.Models;
+using Watchman.Discord.Areas.Protection.Services;
+
+namespace Watchman.Discord.Areas.Protection.Strategies
+{
+    public class SpamPunishmentStrategy : ISpamPunishmentStrategy
+    {
+        private readonly IPunishmentsCachingService _punishmentsCachingService;
+
+        public SpamPunishmentStrategy(IPunishmentsCachingService punishmentsCachingService)
+        {
+            _punishmentsCachingService = punishmentsCachingService;
+        }
+
+        public Punishment GetPunishment(ulong userId, SpamProbability spamProbability)
+        {
+            var takeFromTime = DateTime.Now.AddHours(-12);
+            var warnsCount = _punishmentsCachingService.GetUserWarnsCount(userId, takeFromTime);
+
+            var punishmentOption = spamProbability switch
+            {
```

Comment:
hmm
może da sie tu zastosować jakiś prosty trick z mnożeniem * 2 
(enumy to wielokrotności 2)

Reply:
nie za bardzo wiem jak by to zrobić z mnożeniem
trzeba jeszcze uwzględniać warnsCount